src/nunit.analyzers/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -39,5 +40,20 @@ internal static class IMethodSymbolExtensions
 
             return new Tuple<uint, uint, uint>(requiredParameters, optionalParameters, paramsParameters);
         }
+
+        /// <summary>
+        /// Returns true if method is implementation of method in interface
+        /// </summary>
+        internal static bool IsInterfaceImplementation(this IMethodSymbol @this, string interfaceFullName)
+        {
+            var interfaceType = @this.ContainingType.AllInterfaces.FirstOrDefault(i => i.GetFullMetadataName() == interfaceFullName);
+
+            if (interfaceType == null)
+                return false;
+
+            return interfaceType.GetMembers().OfType<IMethodSymbol>()
+                .Any(interfaceMethod => interfaceMethod.Name == @this.Name
+                    && @this.ContainingType.FindImplementationForInterfaceMember(interfaceMethod).Equals(@this));
```

Comment:
Good catch! 
I don't think it works for overridden methods, but don't think that's the issue here. Finding overridden method definition is much easier (there is property `OverriddenMethod` in `IMethodSymbol`) than implemented interface definition (I assume because same method can be an implementation for multiple interfaces)